psp/models/recent_history.py

```diff
@@ -164,18 +169,25 @@ def __init__(
         """
         super().__init__(config)
         # Validate some options.
-        if nwp_dropout > 0.0 or pv_dropout > 0.0:
+
+        if nwp_dropout > 0.0 or pv_dropout > 0.0 or excarta_dropout > 0.0:
             assert random_state is not None
 
         if use_nwp:
             assert nwp_data_source is not None
 
+        if use_excarta:
+            assert excarta_data_source is not None
```

Comment:
If doing the changes above, could change this more to asserting that the list of NWP data sources matches the length of the list of nwp variables, and nwp dropout